pkg/dependency/parser/nodejs/packagejson/parse.go

```diff
@@ -40,6 +43,10 @@ func (p *Parser) Parse(r io.Reader) (Package, error) {
 		return Package{}, xerrors.Errorf("JSON decode error: %w", err)
 	}
 
+	if !IsValidName(pkgJSON.Name) {
+		return Package{}, xerrors.Errorf("Name can only contain URL-friendly characters")
```

Comment:
I think the warning message is enough. Otherwise, if a package.json file contains an invalid name for testing, it fails the whole scanning.

Reply:
Or is it anyway converted into a warning in analyzer.go?

Reply:
> Or is it anyway converted into a warning in analyzer.go

Right.